src/pam_hbac_obj.c

```diff
@@ -61,6 +64,10 @@ getgroupname(gid_t gid)
         return NULL;
     }
 
+#ifndef getgrgid_r               // to avoid not compiling on hpux
+  extern int getgrgid_r(gid_t, struct group *, char *, size_t, struct group**);
+#endif
+
```

Comment:
Hmm, could you please share config.log and the compilation error messages? Does HP-UX not have `getgrgid_r` at all? Does it at least have the non-reentrant `getgrgid`?